### PR TITLE
feat(ServerApplication): termination callback #4643

### PR DESCRIPTION
**Pull request by aleks-f**

### CODE REVIEW

File: Util/include/Poco/Util/ServerApplication.h [hunk 1]
```diff
@@ -122,8 +123,21 @@ class Util_API ServerApplication: public Application
...
+	/// 
+	///     static const std::string cbMsg = "Termination custom message"s;
+	///     auto tCB = [](const std::string& message)
+	///     {
+	///         std::cout << message << std::endl;
+	///     };
+	///     ServerApplication::registerTerminateCallback(tCB, cbMsg);
 {
 public:
+	using TerminateCallback = std::function<void(const std::string&)>;
```
Comment (matejk) [inline Util/include/Poco/Util/ServerApplication.h:139]: What is the purpose of the `std::string` argument?

File: Util/include/Poco/Util/ServerApplication.h [hunk 2]
```diff
@@ -158,6 +172,11 @@ class Util_API ServerApplication: public Application
 		/// waitForTerminationRequest(), this method will return
 		/// and the application can shut down.
 
+	static void registerTerminateCallback(TerminateCallback tCB,
+		const std::string& message = _terminateMessage);
+		/// Registers a termination callback.
+		/// Used to notify all interested users about system shutdown.
```
Comment (matejk) [inline Util/include/Poco/Util/ServerApplication.h:178]: The description shall be rephrased IMO. The callback can perform any cleanup or whatever is required to do on shutdown, notifying users is only one of the possibilities.

**Review by matejk — APPROVED** [review]
A few comments to consider.

File: Util/include/Poco/Util/ServerApplication.h (hunk 1 above)
Comment (aleks-f) [reply Util/include/Poco/Util/ServerApplication.h:139]: I guess whatever is needed in there can just be captured by lambda

**Pull request merged**